### PR TITLE
fix: resolve the issue with Kubernetes upgrade

### PR DESCRIPTION
**Pull request by AlekSi**
Add missing cases, refactoring.

- [ ] you linked an issue (if applicable)
- [x] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

### CODE REVIEW

**Review by smira — APPROVED** [review]

**Comment by AlekSi** [comment]
`make lint` fails for what seems to be unrelated

File: pkg/cluster/kubernetes/upgrade.go [hunk 1]
```diff
@@ -5,6 +5,9 @@
 package kubernetes
 
 import (
+	"fmt"
+
+	"github.com/coreos/go-semver/semver"
```
Comment (AlekSi) [inline pkg/cluster/kubernetes/upgrade.go:10]: We already had https://github.com/coreos/go-semver, https://github.com/blang/semver, and https://github.com/hashicorp/go-version in our deps. I picked the first one as it seems to least likely to go away.

**Pull request merged**